test/spmd/test_dynamo_spmd.py

```diff
@@ -15,16 +15,22 @@
 
 class SimpleLinear(nn.Module):
 
-  def __init__(self):
+  def __init__(self, mesh=None):
     super(SimpleLinear, self).__init__()
     self.fc1 = nn.Linear(128, 128)
     self.relu = nn.ReLU()
     self.fc2 = nn.Linear(128, 1)
     # Add an additional 1x1 layer at the end to ensure the final layer
     # is not sharded.
     self.fc3 = nn.Linear(1, 1)
+    # If mesh is not none, we'll do a mark sharding inside the forward function
+    # to ensure dynamo can recognize and trace it in a torch compile.
+    self.mesh = mesh
 
   def forward(self, x):
+    if self.mesh and 'xla' in str(self.fc2.weight.device):
```

Comment:
@yeounoh, I updated the unit test for doing `mark_sharding` inside torch compile to be part of the existing `SimpleLinear`. Here, I just do a `mark_sharding` call inside the `forward` function. Please let me know if you think this will suffice.

Reply:
SG